azurelinuxagent/common/protocol/hostplugin.py

```diff
@@ -480,8 +480,9 @@ def format_message(msg):
                 logger.info(message)
                 add_event(op=WALAEventOperation.HostPlugin, message=message, is_success=True)
 
-            # Don't support HostGAPlugin versions older than 115
-            if vm_settings.host_ga_plugin_version < FlexibleVersion("1.0.8.115"):
+            # Don't support HostGAPlugin versions older than 123
+            # TODO: update the minimum version to 1.0.8.117 before release
+            if vm_settings.host_ga_plugin_version < FlexibleVersion("1.0.8.117"):
                 raise_not_supported(reset_state=True)
```

Comment:
If not supported, how does this update `supports_fast_track` flag as `false`

Reply:
the flag is maintained by the UpdateHandler and computed like this

`supports_fast_track = self._goal_state.extensions_goal_state.source_channel == GoalStateChannel.HostGAPlugin
`

If the HostGAPlugin does not support vmsettings (or is the wrong version) we fallback to the WireServer

Reply:
Note also that the HostGAPlugin may support fast track, but the feature may be disabled in waagent.conf. The above condition takes care of that too.

The naming is a little awkward: CRP needs FastTrack to be added to the **supported**Features, but what the flag really means is that the agent is requesting CRP to start using FastTrack. Of course, if the agent sets this flag, it must support FT. But the agent could support FT and still not request CRP to use it (e.g. the user disabled the feature, or the hostgaplugin is not the correct version).

Reply:
got you. Thanks for explaining.

Reply:
I am doing another pass of the code (separate PRs) and will add comments and tests to explain the behavior of the feature, as well as to address the TODOs (the behavior is already described in the spec for fast track on the linux agent, but i should capture that in tests)